### ISSUE TITLE
[Bug]: LayoutEngine rect not respected when saving figure with `rcParams['figure.constrained_layout.use'] = True`

### ISSUE THREAD

**Issue by s-bear** [matplotlib/matplotlib#24389]
### Bug summary

I use the `rect` parameter of the figure's layout engine to set figure margins. For figures with `layout='constrained'` this parameter appears to be ignored if `rcParams['figure.constrained_layout.use']` is `True` when saving the figure. 

### Code for reproduction

```python
import matplotlib.pyplot as plt
plt.rcParams['figure.constrained_layout.use']=True
fig, ax = plt.subplots()
engine = fig.get_layout_engine()
engine.set(rect=[0.1,0.1,0.8,0.8]) # (left, bottom, width, height)
engine.execute(fig)
fig.savefig('test-constrained-True.png')
```

### Actual outcome

The saved figure does not have the expected layout. It does not match the figure displayed interactively. The axes extend to the edges of the figure area.
![test-constrained-True](https://user-images.githubusercontent.com/4016348/200223680-cb1cba6b-ee61-4533-94f4-5b0f46441ca9.png)

### Expected outcome

The saved figure has matches what is shown interactively. The axes do not extend to the edges of the figure area, respecting the `rect` parameter. This saved figure is generated with the code, which does not demonstrate the bug:
```python
plt.rcParams['figure.constrained_layout.use'] = False
fig, ax = plt.subplots(layout='constrained')
engine = fig.get_layout_engine()
engine.set(rect=[0.1,0.1,0.8,0.8]) # (left, bottom, width, height)
engine.execute(fig)
fig.savefig('test-constrained-False.png')
```
![test-constrained-False](https://user-images.githubusercontent.com/4016348/200223804-509d7fec-37a0-41fb-aede-541852706cbf.png)

### Operating system

Windows

### Matplotlib Version

3.6.1

### Matplotlib Backend

Qt5Agg

### Python version

3.10.6

### Jupyter version

n/a

### Installation

conda

**Comment by jklymak** [comment]
What are you trying to accomplish by setting the rcParam?  I'm surprised this has any effect on existing figures.  If you want to turn constrained layout on or off I'd just do so manually.

**Comment by s-bear** [comment]
I set it in the rcParams because I wanted to use constrained layout by default for all figures in a script. Perhaps my example isn't completely minimal. If the rcParam is set at all, even if the saved figure appears differently from the interactive one.


Samuel B. Powell, PhD

Postdoctoral Research Fellow



Queensland Brain Institute

The University of Queensland

Building 79, Upland Rd

Brisbane, QLD 4072, Australia



Office: +61 7 3365 4089<tel:+61733654089> Mobile: +61 0432 708 829<tel:+61432708829>

Email: ***@***.******@***.***> Web: http://www.uq.edu.au/ecovis

________________________________

**Comment by s-bear** [comment]
I just edited the bug report to make the example code more consise and demonstrate the issue more clearly. Cheers, Sam

**Comment by jklymak** [comment]
This was incorrect changing of the layout engine during the print stage, and was fixed in https://github.com/matplotlib/matplotlib/pull/24167 which was released with 3.6.2.  Sorry this broke. Essentially a duplicate of #24133. 

BTW, you should not need the `engine.execute(fig)` step as that gets called upon either `show` or `savefig`

**Comment by tacaswell** [comment]
I can confirm that this is fixed in 3.6.2.

Thank you for the very clear example @s-bear !
